include/envoy/server/filter_config.h

```diff
@@ -58,6 +59,11 @@ class FactoryContext {
   virtual bool healthCheckFailed() PURE;
 
   /**
+   * @return HttpRouteManager& singleton for use by the entire server.
+   */
+  virtual HttpRouteManager& httpRouteManager() PURE;
```

Comment:
@mattklein123 I have changed the HttpRouteManager interface like we talked about to only expose `get` and `list` the configured route providers. 

What is the pattern to follow here to not expose the entire manager? is it to have a `getRouteConfigProvider` method in this interface, or now that we can only get and list, is it fine to expose the manager at the context level?